src/Sarif/Visitors/InsertOptionalDataVisitor.cs

```diff
@@ -141,13 +141,15 @@ public override FileData VisitFileData(FileData node)
                         }
 
                         int length = node.Length;
-                        node = FileData.Create(uri, _dataToInsert, node.MimeType, encoding: encoding);
+                        node = Artifact.Create(uri, _dataToInsert, node.MimeType, encoding: encoding);
                         node.Length = length;
+                        fileLocation.Index = -1;
+                        node.Location = fileLocation;
```

Comment:
>Location [](start = 29, length = 8)

On looking at some test file delta, i noticed this code wasn't migrating artifact location data to the artifact object. we don't need the index in this case, of course.